apps/aehttp/test/aehttp_ws_test_utils.erl

```diff
@@ -120,20 +120,27 @@ init(WaitingPid) ->
 
 onconnect(_WSReq, Regs) ->
     ct:log("Ws connected"),
-    [WaitingPid] = get_registered(waiting_connected, Regs),
-    Regs1 = delete_registered(WaitingPid, waiting_connected, Regs),
-    inform_registered(WaitingPid, websocket, connected),
-    {ok, Regs1}.
+    self() ! ping,
+    {ok, Regs}.
 
 
 ondisconnect({error, {400, <<"Bad Request">>}}, Regs) ->
     {close, normal, Regs};
 ondisconnect({remote, closed}, Regs) ->
     ct:log("Connection closed, closing"),
-    {close, "closed", Regs}.
-
-websocket_handle({pong, _}, _ConnState, Regs) ->
-    {ok, Regs};
+    {close, normal, Regs}.
+
+websocket_handle({pong, Nonce}, _ConnState, Regs) ->
+    case get_registered(waiting_nonce, Regs) of
+        [Nonce] ->
+            Regs1 = delete_registered(Nonce, waiting_nonce, Regs),
+            [WaitingPid] = get_registered(waiting_connected, Regs1),
+            inform_registered(WaitingPid, websocket, connected),
```

Comment:
I see this change makes the WS considered connected after additional ping-pong initiated by client. It is ok. I wonder whether this fixes a symptom or race condition?

I read cowboy supports ping:
> Note that ping and pong frames require no action from the handler as Cowboy will automatically reply to ping frames.

Reply:
@lucafavatella It's bc of a change in cowboy websocket behaviour. In cowboy's 2.* version the handshake is done before the return on `websocket_init/1` hence the client's execute `onconnect/2` without any delays (`jobs:ask/1`).